native/source/mixpanel/detail/worker.cpp

```diff
@@ -190,6 +198,46 @@ namespace mixpanel
             }
         }
 
+        int Worker::parse_www_retry_after(const nanowww::Response& response)
+        {
+            mixpanel->log(Mixpanel::LogEntry::LL_TRACE, "/track HTTP Response Headers: \n" + response.headers()->as_string());
+            mixpanel->log(Mixpanel::LogEntry::LL_TRACE, "/track HTTP Response Body: \n" + response.content());
+
+            // Check for an HTTP Retry-After header
+            auto retry_after_header = response.get_header("Retry-After");
+            auto retry_after = 0;
+            if (!retry_after_header.empty()) {
+                retry_after = atoi(retry_after_header.c_str());
+                mixpanel->log(Mixpanel::LogEntry::LL_TRACE, "Retry After Header: " + retry_after_header);
+            }
+
+            // Check for a 5XX response code
+            bool failed = (500 <= response.status() && response.status() <= 599);
+            if (failed) {
+                failure_count++;
+                mixpanel->log(Mixpanel::LogEntry::LL_ERROR, "/track HTTP Call Failed - Status Code (" + std::to_string(response.status()) + "): " + response.content());
+            } else {
+                failure_count = 0;
+            }
+
+            // Calculate exponential back off
+            if (failure_count > 1) {
+                retry_after = std::max(retry_after, calculate_back_off_time(failure_count));
+                mixpanel->log(Mixpanel::LogEntry::LL_INFO, "Adding exponential back off time of " + std::to_string(retry_after) + " seconds.");
+            }
+
+            auto allowed_after_time = time(0) + retry_after;
```

Comment:
Please add above this line something like

    auto now = time(0);

and then `s/time(0)/now/` in all the subsequent lines.